### PR TITLE
add 141. Linked List Cycle

### PR DESCRIPTION
**Pull request by nayuta-ai**

### CODE REVIEW

File: 141_Linked_List_Cycle.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+class Solution:
+    def hasCycle(self, head: ListNode) -> bool:
+        fast = head
+        slow = head
+        while fast and fast.next:
+            fast = fast.next.next
+            slow = slow.next
+            if fast==slow:
+                return True
```
Comment (tkkawa) [inline 141_Linked_List_Cycle.py:15]: Leave one line after "return"

File: 141_Linked_List_Cycle.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#         self.next = None
+
+class Solution:
+    def hasCycle(self, head: ListNode) -> bool:
+        fast = head
+        slow = head
+        while fast and fast.next:
+            fast = fast.next.next
+            slow = slow.next
+            if fast==slow:
```
Comment (tkkawa) [inline 141_Linked_List_Cycle.py:14]: Insert a space before and after ==

**Review by tkkawa — CHANGES_REQUESTED** [review]
write time and space complexity

File: 141_Linked_List_Cycle.py (hunk 1 above)
Comment (nayuta-ai) [reply 141_Linked_List_Cycle.py]: leave this line

File: 141_Linked_List_Cycle.py (hunk 2 above)
Comment (nayuta-ai) [reply 141_Linked_List_Cycle.py]: Insert a space in line 14

**Review by tkkawa — CHANGES_REQUESTED** [review]
write both time complexity and space complexity

File: 141_Linked_List_Cycle.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Definition for singly-linked list.
+# class ListNode:
+#     def __init__(self, x):
+#         self.val = x
+#         self.next = None
+# time O(n)
+# space O(n)
```
Comment (tkkawa) [inline 141_Linked_List_Cycle.py:7]: O(1)

**Review by tkkawa — CHANGES_REQUESTED** [review]

**Review by tkkawa — APPROVED** [review]
LGTM

**Pull request merged**